pkg/apis/serving/v1alpha2/predictor.go

```diff
@@ -79,6 +83,36 @@ func (p *PredictorSpec) Validate(config *InferenceServicesConfig) error {
 	if err := validateReplicas(p.MinReplicas, p.MaxReplicas); err != nil {
 		return err
 	}
+	if err := validateResourceRequirements(predictor.GetResourceRequirements()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func transferToCoreResourceRequirements(rr *v1.ResourceRequirements) *core.ResourceRequirements {
```

Comment:
can we move this function to utils ? looks like it is shared among predictor, transformer and explainer.

Reply:
>> I meant to put the validation logic to a common place but I saw there's validateStorageURI function which is used by explainer too. I think it's fine since it's in same package so will follow.

See above but sure I can move it together with validateStorageURI function to a new util.go